ReactWindows/ReactNative.Shared/ReactNativeHost.cs

```diff
@@ -8,94 +8,15 @@
 namespace ReactNative
 {
     /// <summary>
-    /// Simple class that holds the <see cref="ReactInstanceManager"/>. 
+    /// Simple class that is responsible for the creation of one <see cref="ReactRootView"/>. 
```

Comment:
>t is responsible for the creation of one <see cref="ReactRootView"/>.  [](start = 24, length = 70)

I'm not sure I agree with this. All apps are going to have a "main" window. I think the ReactNativeHost holds the MainComponentName purely for convenience. I don't think it has to be associated with all React roots.

Reply:
I understand where you're coming from, it's just that there's one scenario where it's the "non main" window that starts first.
The "My People" one. It's always mentioned when something obscure has to happen.

A user opening a "my people" panel when the main app is not running will instantiate the root view for that thing rather than the main root view.
I'm open to other solutions, but it would have to address this scenario somehow.

---
In reply to: [165947212](https://github.com/Microsoft/react-native-windows/pull/1615#discussion_r165947212) [](ancestors = 165947212)

Reply:
I'm confused why this class has any impact on the order of windows starting. The "choreography" of windows should be within the control of App.xaml.cs, when you get an activation to start with the "My People" app, you don't have to use the existing `OnCreate` extension method for `ReactNativeHost` - you could add a new extension method for `OnCreate` that takes the component name as a parameter. Maybe I'm missing something...

---
In reply to: [166092692](https://github.com/Microsoft/react-native-windows/pull/1615#discussion_r166092692) [](ancestors = 166092692,165947212)

Reply:
Ok, I could try. I considered RootView class (with its OnCreate, StarReactApplication, etc. ) as somehow hidden from the app, but if app keeps references to them / controls creation order / tweaks the OnCreate sequence, then yes, the library could be kept simpler. Will do.

---
In reply to: [166258284](https://github.com/Microsoft/react-native-windows/pull/1615#discussion_r166258284) [](ancestors = 166258284,166092692,165947212)

Reply:
@rozele fixed it! Much simpler now :).

---
In reply to: [166357622](https://github.com/Microsoft/react-native-windows/pull/1615#discussion_r166357622) [](ancestors = 166357622,166258284,166092692,165947212)